go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -724,13 +726,45 @@ func doMigrateConfig(cmd *cobra.Command, args []string) {
 
 	// If a node has `consensus.validator` set to true and it does not have any runtimes configured,
 	// the new `mode` should be set to `validator`.
+	var isValidator bool
 	if newValidator, ok := m(newCfg["consensus"])["validator"]; ok {
-		isValidator, _ := newValidator.(bool)
+		isValidator, _ = newValidator.(bool)
 		if _, hasRuntimes := m(newCfg["runtime"])["paths"]; !hasRuntimes && isValidator {
 			nodeMode = "validator"
 			delete(m(newCfg["consensus"]), "validator")
 		}
 	}
+	// If node is a validator, it now requires external P2P addresses to have set.
+	if isValidator {
+		// Set P2P registration addresses based on consensus.external_address.
+		ea := m(newCfg["consensus"])["external_address"]
+		if eaStr, ok := ea.(string); !ok {
+			logger.Warn("consensus.external_address missing, not configuring p2p registration parameters", "address", ea)
+		} else {
+			// Parse host from the external address.
+			url, err := url.Parse(eaStr)
+			if err != nil {
+				logger.Error("failed to parse URI from consensus.external_address", "err", err)
+				os.Exit(1)
+			}
+			// Set P2P port if not set.
+			p := url.Port()
+			if _, ok := m(newCfg["p2p"])["port"]; !ok && p != "" {
+				// Port to integer.
+				if pp, err := strconv.Atoi(p); err == nil {
+					m(newCfg["p2p"])["port"] = pp
```

Comment:
Port cannot be the same as the CometBFT P2P port as this is serving a different protocol and such a configuration will not work. If the port is not set, I would use a default one. Also make sure that the `p2p.port` is used in `p2p.registration.addresses` by default (and not the one from `external_address`).

Reply:
Makes sense, I misunderstood the issue description. I guess 9200 makes sense as a default (similar to known seed)?